src/goto-instrument/contracts/dynamic-frames/dfcc_infer_loop_assigns.cpp

```diff
@@ -154,10 +157,75 @@ std::unordered_set<irep_idt> gen_loop_locals_set(
   return loop_locals;
 }
 
-assignst dfcc_infer_loop_assigns(
+/// Append all identifiers in `expr` into `identifiers`.
+static void
+append_ids_in_expr(const exprt &expr, std::unordered_set<irep_idt> &identifiers)
+{
+  auto ids = find_symbol_identifiers(expr);
+  identifiers.insert(ids.begin(), ids.end());
+}
+
```

Comment:
No need for this: just use `find_symbols(expr, identifiers);` in any place where you are currently using `append_ids_in_expr`.